### PR TITLE
Add integration tests for resident idp config urls

### PR DESCRIPTION
**Pull request by GANGANI**
### Proposed changes in this pull request
Add integration tests to test resident idp inbound authentication config URLs and inbound provisioning config URLs in super tenant legacy URL mode and tenant legacy URL mode.

### When should this PR be merged
This PR should be merged after merging https://github.com/wso2/carbon-identity-framework/pull/2946 and https://github.com/wso2/product-is/pull/8374

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/ResidentIDPConfigsTestCase.java [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+                "https://localhost:9853/wso2/scim/Users",
+                "Expected SCIM User Endpoint is not found in the tenant mode");
+        Assert.assertEquals(provisioningConfigTenantMap.get("scim2UserEndpoint"),
+                "https://localhost:9853/t/wso2.com/wso2/scim2/Users",
+                "Expected SCIM 2.0 User Endpoint is not found in the tenant mode");
+        Assert.assertEquals(provisioningConfigTenantMap.get("scim2GroupEndpoint"),
+                "https://localhost:9853/t/wso2.com/wso2/scim2/Groups",
+                "Expected SCIM 2.0 Group Endpoint is not found in the tenant mode");
+    }
+}
```
Comment (ivantha) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/ResidentIDPConfigsTestCase.java:262]: Newline :)

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/ResidentIDPConfigsTestCase.java [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+        }
+
+        Assert.assertEquals(provisioningConfigTenantMap.get("scimGroupEndpoint"),
+                "https://localhost:9853/wso2/scim/Groups",
+                "Expected SCIM Group Endpoint is not found in the tenant mode");
+        Assert.assertEquals(provisioningConfigTenantMap.get("scimUserEndpoint"),
+                "https://localhost:9853/wso2/scim/Users",
+                "Expected SCIM User Endpoint is not found in the tenant mode");
+        Assert.assertEquals(provisioningConfigTenantMap.get("scim2UserEndpoint"),
+                "https://localhost:9853/t/wso2.com/wso2/scim2/Users",
```
Comment (ashensw) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/ResidentIDPConfigsTestCase.java:256]: Is this URL correct? This should be ```https://localhost:9853/t/wso2.com/scim2/Users```?

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/ResidentIDPConfigsTestCase.java [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+                "Expected SCIM 2.0 User Endpoint is not found in the tenant mode");
+        Assert.assertEquals(provisioningConfigTenantMap.get("scim2GroupEndpoint"),
+                "https://localhost:9853/t/wso2.com/wso2/scim2/Groups",
```
Comment (ashensw) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/ResidentIDPConfigsTestCase.java:259]: Same as above.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/ResidentIDPConfigsTestCase.java [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+
+        Map<String, String> provisioningConfigTenantMap = new HashMap<String, String>();
+        for (ProvisioningConnectorConfig config : provisioningConfigsTenant) {
+            for (Property property: config.getProvisioningProperties()) {
+                provisioningConfigTenantMap.put(property.getName(), property.getValue());
+            }
+        }
+
+        Assert.assertEquals(provisioningConfigTenantMap.get("scimGroupEndpoint"),
+                "https://localhost:9853/wso2/scim/Groups",
```
Comment (ashensw) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/ResidentIDPConfigsTestCase.java:250]: This is not tenant supported?

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/ResidentIDPConfigsTestCase.java (hunk 2 above)
Comment (GANGANI) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/ResidentIDPConfigsTestCase.java:256]: This is correct. for SCIM 2.0 User Endpoint URL is `https://localhost:9443/t/wso2.com/wso2/scim2/Users`

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/ResidentIDPConfigsTestCase.java (hunk 4 above)
Comment (GANGANI) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/ResidentIDPConfigsTestCase.java:250]: No. Only SCIM 2.0 endpoints have tenant domain in path.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/ResidentIDPConfigsTestCase.java [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+import org.wso2.carbon.identity.application.common.model.idp.xsd.Property;
+import org.wso2.carbon.identity.application.common.model.idp.xsd.ProvisioningConnectorConfig;
+import org.wso2.carbon.integration.common.admin.client.AuthenticatorClient;
+import org.wso2.identity.integration.common.clients.Idp.IdentityProviderMgtServiceClient;
+import org.wso2.identity.integration.common.utils.ISIntegrationTest;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class ResidentIDPConfigsTestCase extends ISIntegrationTest {
```
Comment (mefarazath) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/ResidentIDPConfigsTestCase.java:37]: Let's have a class comment

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/ResidentIDPConfigsTestCase.java [hunk 6]
```diff
@@ -0,0 +1,262 @@
...
+                "https://localhost:9853/wso2/scim/Users",
+                "Expected SCIM User Endpoint is not found");
+        Assert.assertEquals(provisioningConfigMap.get("scim2UserEndpoint"),
+                "https://localhost:9853/wso2/scim2/Users",
+                "Expected SCIM 2.0 User Endpoint is not found");
+        Assert.assertEquals(provisioningConfigMap.get("scim2GroupEndpoint"),
+                "https://localhost:9853/wso2/scim2/Groups",
+                "Expected SCIM 2.0 Group Endpoint is not found");
+
+        ConfigurationContext configContext = ConfigurationContextFactory.createConfigurationContextFromFileSystem
```
Comment (mefarazath) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/ResidentIDPConfigsTestCase.java:155]: Can we move the tenant case to a new test case?

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/ResidentIDPConfigsTestCase.java (hunk 2 above)
Comment (mefarazath) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/ResidentIDPConfigsTestCase.java:256]: I think its a bug :)
we need to fix it and change the tests as well

Comment (GANGANI) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/ResidentIDPConfigsTestCase.java]: Bug was fixed in PR: https://github.com/wso2/carbon-identity-framework/pull/2946 and tests were changed in https://github.com/wso2/product-is/pull/8370/commits/f196fdb5ff9850329b87aaffb483f2d68783cb04

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/ResidentIDPConfigsTestCase.java (hunk 3 above)
Comment (GANGANI) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/ResidentIDPConfigsTestCase.java]: Fixed https://github.com/wso2/product-is/pull/8370/commits/f196fdb5ff9850329b87aaffb483f2d68783cb04

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/ResidentIDPConfigsTestCase.java (hunk 6 above)
Comment (GANGANI) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/ResidentIDPConfigsTestCase.java:162]: Done in https://github.com/wso2/product-is/pull/8370/commits/f196fdb5ff9850329b87aaffb483f2d68783cb04

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/ResidentIDPConfigsTestCase.java (hunk 5 above)
Comment (GANGANI) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/ResidentIDPConfigsTestCase.java:40]: Added in https://github.com/wso2/product-is/pull/8370/commits/f196fdb5ff9850329b87aaffb483f2d68783cb04

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/ResidentIDPConfigsTestCase.java [hunk 7]
```diff
@@ -0,0 +1,269 @@
...
+
+        super.init();
+        ConfigurationContext configContext = ConfigurationContextFactory.createConfigurationContextFromFileSystem(null, null);
+        idpMgtServiceClient = new IdentityProviderMgtServiceClient(sessionCookie, backendURL, configContext);
+    }
+
+    @Test(groups = "wso2.is", description = "Test resident IdP config URLs in super tenant mode")
+    public void testResidentIdPConfigs() throws Exception {
+
+        log.info("Retrieving resident identity provide");
```
Comment (dewniMW) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/ResidentIDPConfigsTestCase.java:56]: `provide` -> `provider`

Comment (GANGANI) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/ResidentIDPConfigsTestCase.java:56]: This code was changed with https://github.com/wso2/product-is/commit/a706b3711c3559a331e7d18c1d30746e787b70f5

**Pull request merged**